### PR TITLE
docs: add library starter template docs

### LINKED ISSUE #2907: Create a template for developing Remotion libraries + components
Create a template that allows one to develop a library in the Remotion ecosystem with the best practices applied out of the box.

Should be a monorepo (for example with pnpm, same as this repository) which includes two packages:

- A simple Remotion library that as a starter, includes a simple function (e.g. `useSeconds()`).
- Should include a sample project with the Remotion Studio so that one can test the function
- It should package the library as CJS+ESM for maximum compatibility (for example using tsup)
- The template should be allowed to live in the remotion-dev GitHub organization (while crediting you as the author)
- There should be a README or doc site that explains how to use the template (clone, start, change name, publish to npm)

## Acceptance criteria

Deliver a repository that one can clone as a template and develop their own Remotion function / component and publish to NPM as well as test it in a Remotion Studio environment.

## Bounty 

💎 This issue is part of Hacktoberfest and has a bounty on it!
* **General Rules**: https://www.remotion.dev/bounties
- **How to contribute**: https://www.remotion.dev/docs/contributing/
- **Implementing a new feature**: https://www.remotion.dev/docs/contributing/feature
- **Implementing a new option**: https://www.remotion.dev/docs/contributing/option
- **Write documentation**: https://www.remotion.dev/docs/contributing/docs

🎃 **Hacktoberfest rules**
- You can only work on 1 issue at a time.
- Only 1 person can work on the same issue.
- We explicitly assign a developer. Don’t work on an issue if we do not assign you.
- First-time contributors will get preferential treatment.
- You must set up the Remotion repository locally and test your changes. [Contributing to Remotion | Remotion | Make videos programmatically in React](https://www.remotion.dev/docs/contributing)
- Your PR must fulfill the acceptance criteria.
- If you use AI, you must verify that the output is correct.
- All tests (functionality and formatting) must pass.

Issue already taken? Find more issues under https://remotion.dev/bounties

/bounty 300

### PR DESCRIPTION
**Pull request by Just-Moh-it**
...

Still might need github actions and automations for things like auto publish to npm, and safeguards like hooks used outside player, but they're out of the scope for this pr

(current remotion repository nomenclature suggests it would be named remotion-dev/library-template, so added github links for that until the repo is transferred :)

closes #2907 
/attempt #2907

### CODE REVIEW

File: packages/bugs/package.json [hunk 1]
```diff
@@ -1,4 +1,5 @@
 {
+  "name": "bugs",
```
Comment (Just-Moh-it) [inline packages/bugs/package.json:2]: Had errors with using `pnpm install` at repo root... so used bun but go an error that said bugs package didn't have a name, so just added it

File: packages/docs/docs/contributing/library.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+---
+
+Building libraries and components for the remotion ecosystem is a great way to contribute to remotion.
+
+We suggest starting from Remotion's library/component starter template which has all the best practices to create libraries and components for remotion.
+
+## See also
+
+- [NPM Kiosk](https://remotion.pro/npm-kiosk) to sell your npm package
+- [Remotion Store](https://remotion.pro/) to list your package for sale
```
Comment (Just-Moh-it) [inline packages/docs/docs/contributing/library.md]: (potential upsells lol 😎)

Comment (JonnyBurger) [reply packages/docs/docs/contributing/library.md]: love it

File: packages/bugs/package.json (hunk 1 above)
Comment (JonnyBurger) [reply packages/bugs/package.json:2]: oh interesting, thanks for adding that

**Comment by JonnyBurger** [comment]
Let's go 🙌🏼

**Pull request merged**